src/lib/logger/impl.ml

```diff
@@ -76,9 +76,11 @@ module Metadata = struct
 
   let extend (t : t) alist =
     List.fold_left alist ~init:t ~f:(fun acc (key, data) ->
-        String.Map.add_exn acc ~key ~data )
+        String.Map.set acc ~key ~data )
 end
 
+let global_metadata = ref []
```

Comment:
This is a really minor nit, but just for future sanity sake, can we change the interface into adding things to `global_metadata` so that it's append only? As in, don't expose this ref, but just expose a function `add_to_global_metadata` (or a better named version of this) which can be called with a list of new metadata to add.